### PR TITLE
Use libxmlb bound parameters to speed up te device verification

### PR DESCRIPTION
**Pull request by hughsie**
This means we only parse the complicated xpath query once, rather than for
every GUID the device has.

The code flow is also simplified and split out into two functions.

### CODE REVIEW

File: src/fu-engine.c [hunk 1]
```diff
@@ -785,6 +785,106 @@ fu_engine_get_component_by_guids (FuEngine *self, FuDevice *device)
...
+			     "failed to find %s", fn);
+		return NULL;
+	}
+
+	if (!xb_builder_source_load_file (source, file,
+					  XB_BUILDER_SOURCE_FLAG_NONE,
+					  NULL, error))
+		return NULL;
+	xb_builder_import_source (builder, source);
+	silo = xb_builder_compile (builder,
```
Comment (superm1) [inline src/fu-engine.c:819]: Isn't this a memory leak?

Comment (superm1) [reply src/fu-engine.c:819]: (specifically because of the call to `xb_silo_new` earlier on with this same pointer)

Comment (hughsie) [reply src/fu-engine.c:819]: Yes, and it was before! Will fix in a smarter way, good catch.

Comment (hughsie) [reply src/fu-engine.c:820]: `g_object_set_data_full` to the rescue. We can't unconditionally ref the silo in the node else we get a dep loop and just leak tons of memory.

**Pull request merged**